### ISSUE TITLE
Have user function return a Result

### ISSUE THREAD

**Issue by sjwiesman** [aljoscha/statefun-rust#7]
Currently, user functions return an `Effect` which means in the case of failure they can only panic. If instead, it returned a `Result<Effect, SomeErrorType>` there would be a more graceful failure scenario.

**Comment by aljoscha** [comment]
Yes, this is a good point. We could either change the existing method or add a new one.

Also, I'm reworking error handling a bit: https://github.com/aljoscha/statefun-rust/pull/9

**Comment by slinkydeveloper** [comment]
This is something that may be handled in https://github.com/aljoscha/statefun-rust/issues/12. The macro, from the return type, can figure out if there's any need to handle the `Result`
